tests/system.rs

```diff
@@ -485,3 +505,230 @@ fn string_eq_ignore_newline_repr_test() {
+        mismatches.insert(PathBuf::from("Configurations.md"), compare);
+        print_mismatches(mismatches, |line_num| {
+            format!(
+                "\nMismatch at Configurations.md:{}:",
+                line_num + self.code_block_start.unwrap() - 1
+            )
+        });
+    }
+
+    fn formatted_has_diff(&self, file_map: FileMap) -> bool {
+        let &(ref _file_name, ref text) = file_map.first().unwrap();
+        let compare = make_diff(self.code_block.as_ref().unwrap(), text, DIFF_CONTEXT_SIZE);
+        if !compare.is_empty() {
+            self.print_diff(compare);
+            return true;
+        }
+
+        false
+    }
+
+    // Return a bool indicating if formatting this code block is an idempotent
+    // operation. This function also triggers printing any formatting failure
+    // messages.
+    fn formatted_is_idempotent(&self) -> bool {
+        // Verify that we have all of the expected information.
+        if !self.code_block_valid() {
+            return false;
+        }
+
+        let input = Input::Text(self.code_block.as_ref().unwrap().to_owned());
+        let config = self.get_block_config();
+
+        let (error_summary, file_map, _report) =
+            format_input::<io::Stdout>(input, &config, None).unwrap();
+
+        !self.has_parsing_errors(error_summary) && !self.formatted_has_diff(file_map)
+    }
+
+    fn extract<I: Iterator<Item = String>>(
+        file: &mut Enumerate<I>,
+        prev: Option<&ConfigCodeBlock>,
+    ) -> Option<ConfigCodeBlock> {
+        let mut code_block = ConfigCodeBlock::new();
+        code_block.config_name = prev.map_or(None, |cb| cb.config_name.clone());
+
+        loop {
+            match ConfigurationSection::get_section(file) {
+                Some(ConfigurationSection::CodeBlock((block, start_line))) => {
+                    code_block.set_code_block(block, start_line);
+                    break;
+                }
+                Some(ConfigurationSection::ConfigName(name)) => {
+                    code_block.set_config_name(Some(name));
+                }
+                Some(ConfigurationSection::ConfigValue(value)) => {
+                    code_block.set_config_value(Some(value));
+                }
+                None => return None, // end of file was reached
+            }
+        }
+
+        Some(code_block)
+    }
+}
+
+// Read Configurations.md and build a `Vec` of `ConfigCodeBlock` structs with one
+// entry for each Rust code block found. Behavior:
+// - Rust code blocks are identifed by lines beginning with "```rust".
+// - One explicit configuration setting is supported per code block.
+// - Rust code blocks with no configuration setting are illegal and cause an
+//   assertion failure.
+// - Configuration names in Configurations.md must be in the form of
+//   "## `NAME`".
+// - Configuration values in Configurations.md must be in the form of
+//   "#### `VALUE`".
+fn get_code_blocks() -> Vec<ConfigCodeBlock> {
```

Comment:
@nrc: I don't expect to use this function elsewhere, which makes me want to limit its scope. My instinct is to nest it inside `configuration_snippet_tests` (below). What do you think about that? Is that a Rusty sort of thing to do? I haven't seen heavy use of nested functions in Rust code (not that I have seen a lot of Rust code).

Reply:
nested functions are fine to use. They are not super-popular because unless the nested function is very small it tends to just make things hard to read.